### PR TITLE
Documentation for fit function DiffRotDiscreteCircle

### LINKED ISSUE #16480: Fit functions without documentation page
Every fit function should have a documentation page: http://docs.mantidproject.org/nightly/fitfunctions/index.html

There is no documentation for the following functions:
I've sectioned these out as best I can based on the history of the .h file.  Please create a pull request for each section.

- [ ] ComptonPeakProfile?, ComptonScatteringCountRate? GaussianComptonProfile? GramCharlierComptonProfile? (there is documentation for MultivariateGaussianComptonProfile)
- [ ] PoldiSpectrumConstantBackground, PoldiSpectrumDomainFunction, PoldiSpectrumLinearBackground, PoldiSpectrumPawleyFunction?
- [x] Polynomial #27412 
- [ ] MultiDomainFunction
- [ ] ReflectivityMulf
- [x] UserFunctionMD (there is documentation for UserFunction). #28328
- [x] FullprofPolynomial #28329
- [x] Bk2BkExpConvPV (but there is documentation for BackToBackExponential) #28331
- [x] ThermalNeutronBk2BkExpAlpha, ThermalNeutronBk2BkExpBeta, , ThermalNeutronBk2BkExpSigma, ThermalNeutronDtoTOFFunction (there is documentation for ThermalNeutronBk2BkExpConvPVoigt) #28330

You won't find the rst pages for these functions under: https://github.com/mantidproject/mantid/tree/master/docs/source/fitting/fitfunctions, so they are not found in the help browser or online pages.

Note that for some of these functions there might be old wiki pages.

For some of these functions (the groups with similar names) it may make sense as there is not much to say, but it would be good to have a documentation page that just states what is the exact difference with respect to the "base" function and link to it (if they are a product or convolution of the base with something else), or link to the important function if they are a component of a larger/more important function.

### LINKED ISSUE #16687: Documentation for fit function DiffRotDiscreteCircle
Includes documentation for the elastic (ElasticDiffRotDiscreteCircle) and inelastic (InelasticDiffRotDiscreteCircle) parts

Adresses in part issue #16480

### PR DESCRIPTION
**Pull request by jmborr**
Documentation for the elastic and inelastic parts of DiffRotDiscreteCircle has been created. It was necessary to extricate the corresponding fitting functions from DiffRotDiscreteCircle.h(.cpp) into their own header and source files.

Fixes #16687 and partially fixes  #16480

To Test: review the documentation and the code, nothing more is required.

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [x] Is user-facing documentation written in a user-friendly manner?
- [x] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by jmborr** [comment]
@FedeMPouzols The doc files for ElasticDiffRotDiscreteCircle and InlasticDiffRotDiscreteCircle are ready. Would you please test this pull request?

**Comment by FedeMPouzols** [comment]
Looks good:
- This moves the code of the individual functions to separate code files so that every of them can have their own documentation page.
- The documentation reads well and its very clear, it provides details and equations as users have been requesting in general
- The documentation builds without any errors/warnings in these files.
- I built this branch locally and the documentation pages display correctly. 

:shipit:

File: docs/source/fitfunctions/DiffRotDiscreteCircle.rst [hunk 1]
```diff
@@ -41,12 +41,15 @@ the two fundamental fitting parameters of the structure factor
    \tau_l^{-1} = 4 \tau^{-1} sin^2(\frac{\pi l}{N})
 
 The transition rate, expressed in units of energy is :math:`h\tau^{-1}`,
-with h = 4.135665616 meV THz.
+with h = 4.135665616 meV ps.
 
-When using InelasticDiffRotDiscreteCircle, he value of Q can be obained either
-though the Q attribute or can be calucated from the input workspace using the
+This function is a composite of `ElasticDiffRotDiscreteCircle <ElasticDiffRotDiscreteCircle>`__ and
```
Comment (FedeMPouzols) [inline docs/source/fitfunctions/DiffRotDiscreteCircle.rst]: I just realized that these links have the same issue as commented in #16730.

**Comment by FedeMPouzols** [comment]
The last commit fixes a couple of links. All looks good as explained [above](https://github.com/mantidproject/mantid/pull/16727#issuecomment-228809267): :shipit:

**Pull request merged**